Source/HTTPHeaders.swift

```diff
@@ -24,51 +24,215 @@
+        
+        return Dictionary(namesAndValues, uniquingKeysWith: { (_, last) in last })
+    }
+}
+
+extension HTTPHeaders: ExpressibleByArrayLiteral {
+    public init(arrayLiteral elements: HTTPHeader...) {
+        self.init()
+        
+        elements.forEach { update(name: $0.name, value: $0.value) }
+    }
+}
+
+extension HTTPHeaders: Sequence {
+    public func makeIterator() -> IndexingIterator<Array<HTTPHeader>> {
+        return headers.makeIterator()
+    }
+}
+
+extension HTTPHeaders: Collection {
+    public var startIndex: Int {
+        return headers.startIndex
+    }
+    
+    public var endIndex: Int {
+        return headers.endIndex
+    }
+    
+    public subscript(position: Int) -> HTTPHeader {
+        return headers[position]
+    }
+    
+    public func index(after i: Int) -> Int {
+        return headers.index(after: i)
+    }
+}
+
+extension HTTPHeaders: CustomStringConvertible {
+    public var description: String {
+        return headers.sorted().map { $0.description }.joined(separator: "\n")
+    }
+}
+
+// MARK: - HTTPHeader
+
+public struct HTTPHeader: Hashable {
+    public let name: String
+    public let value: String
+}
+
+extension HTTPHeader: CustomStringConvertible {
+    public var description: String {
+        return "\(name): \(value)"
+    }
+}
+
+extension HTTPHeader: Comparable {
+    public static func <(lhs: HTTPHeader, rhs: HTTPHeader) -> Bool {
+        return lhs.name < rhs.name
+    }
+}
+
+extension HTTPHeader {
+    public static func acceptLanguage(_ value: String) -> HTTPHeader {
+        return HTTPHeader(name: "Accept-Language", value: value)
+    }
+    
+    public static func acceptEncoding(_ value: String) -> HTTPHeader {
+        return HTTPHeader(name: "Accept-Encoding", value: value)
+    }
+    
+    public static func authorization(_ value: String) -> HTTPHeader {
+        return HTTPHeader(name: "Authorization", value: value)
+    }
+    
+    public static func contentDisposition(_ value: String) -> HTTPHeader {
+        return HTTPHeader(name: "Content-Disposition", value: value)
+    }
+    
+    public static func contentType(_ value: String) -> HTTPHeader {
+        return HTTPHeader(name: "Content-Type", value: value)
+    }
+    
+    public static func userAgent(_ value: String) -> HTTPHeader {
+        return HTTPHeader(name: "User-Agent", value: value)
+    }
+}
 
-extension Dictionary where Key == String, Value == String {
+extension Array where Element == HTTPHeader {
+    func index(of name: String) -> Int? {
+        return firstIndex { $0.name.lowercased() == name.lowercased() }
+    }
+}
+
+// MARK: - Defaults
+
+extension HTTPHeaders {
+    public static let `default`: HTTPHeaders = [.defaultAcceptEncoding,
+                                                .defaultAcceptLanguage,
+                                                .defaultUserAgent]
+    
     public static func authorization(username: String, password: String) -> HTTPHeaders {
```

Comment:
This convenience method should probably be moved to `HTTPHeader`

Reply:
Yes, this is an awkward place, as I ran into in the tests.

Reply:
Changed, and everything is much nicer.